src/Task/ConfigurableTaskTrait.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace Wunderio\GrumPHP\Task;
+
+use GrumPHP\Collection\FilesCollection;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Trait ConfigurableTaskTrait.
+ *
+ * @package Wunderio\GrumPHP\Task
+ */
+trait ConfigurableTaskTrait {
+
+  /**
+   * Name.
+   *
+   * @var string
+   */
+  public $name;
+
+  /**
+   * File separation.
+   *
+   * @var bool
+   */
+  public $isFileSpecific;
+
+  /**
+   * Configurable options.
+   *
+   * @var array[]
+   */
+  public $configurableOptions;
+
+  /**
+   * Configurable options.
+   *
+   * @var array[]
+   */
+  public $configuration;
+
+  /**
+   * Configure Task properties.
+   */
+  public function configure(): void {
```

Comment:
Big methods like this might be more suitable in abstract class, but of course that is not always possible. So, your call :)

Reply:
In this case I would have to repeat same 9 lines for at least 2 times. Also it does make sense that since this trait requires some variables, then it knows how to configure them.